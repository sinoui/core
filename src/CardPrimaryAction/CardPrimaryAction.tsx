import React from 'react';
import styled from 'styled-components';
import { opacify } from 'polished';
import { useRipple } from '@sinoui/ripple';
import OverriableComponent from '../OverridableComponent';
import useMultiRefs from '../utils/useMultiRefs';

export interface Props {
  /**
   * 点击事件
   *
   */
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}

const StyledCardPrimaryAction = styled.div.attrs(() => ({ tabIndex: '0' }))`
  overflow: hidden;
  cursor: pointer;
  outline: none;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 0;
  }

  &:hover {
    ::before {
      background-color: ${({ theme }) =>
        theme.palette.type === 'light'
          ? opacify(-0.96, '#000')
          : opacify(-0.96, '#fff')};
    }
    @media (hover: none) {
      ::before {
        background-color: transparent;
      }
    }
  }

  &:focus {
    background-color: ${({ theme }) =>
      theme.palette.type === 'light'
        ? opacify(-0.88, '#000')
        : opacify(-0.88, '#fff')};
  }

  &:focus::before {
    background-color: ${({ theme }) =>
      theme.palette.type === 'light'
        ? opacify(-0.88, '#000')
        : opacify(-0.88, '#fff')};
  }
`;

/**
 *
 * CardPrimaryAction 主操作区域
 */
const CardPrimaryAction: OverriableComponent<Props, 'div'> = React.forwardRef<
  HTMLDivElement,
  Props
>((props, ref) => {
  const rippleRef = useRipple<HTMLDivElement>();
  const cardPrimaryActionRef = useMultiRefs(rippleRef, ref);
  return <StyledCardPrimaryAction {...props} ref={cardPrimaryActionRef} />;
});

export default CardPrimaryAction;
