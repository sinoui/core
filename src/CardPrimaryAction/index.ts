import CardPrimaryAction from './CardPrimaryAction';

export default CardPrimaryAction;
