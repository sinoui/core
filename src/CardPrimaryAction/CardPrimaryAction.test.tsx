import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import CardPrimaryAction from './CardPrimaryAction';

it('正确渲染CardPrimaryAction', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <CardPrimaryAction>文本</CardPrimaryAction>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
