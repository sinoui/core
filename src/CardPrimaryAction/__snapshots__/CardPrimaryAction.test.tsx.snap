// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正确渲染CardPrimaryAction 1`] = `
<div
<<<<<<< HEAD
  className="sc-AxjAm hAydpj"
=======
  className="CardPrimaryAction__StyledCardPrimaryAction-sc-2ihx33-0 fwVVkG"
>>>>>>> origin/bottomNavigation
  tabIndex="0"
>
  文本
</div>
`;
